velox/core/PlanNode.cpp

```diff
@@ -250,6 +429,28 @@ void ProjectNode::addDetails(std::stringstream& stream) const {
   }
 }
 
+folly::dynamic ProjectNode::serialize() const {
+  auto obj = PlanNode::serialize();
+  obj["names"] = ISerializable::serialize(names_);
+  obj["projections"] = ISerializable::serialize(projections_);
+  return obj;
+}
+
+// static
+PlanNodePtr ProjectNode::create(const folly::dynamic& obj, void* context) {
+  auto source = deserializeSingleSource(obj, context);
+
+  auto names = ISerializable::deserialize<std::vector<std::string>>(
+      obj["names"], context);
+  auto projections = ISerializable::deserialize<std::vector<ITypedExpr>>(
+      obj["projections"], context);
+  return std::make_shared<ProjectNode>(
+      deserializePlanNodeId(obj),
+      std::move(names),
```

Comment:
nit: why not making all these initializations inline? For consistency and readability, you could create a deserializeXxx() method for each member and make this function only:

```
return std::make_shared<ProjectNode>(
      deserializePlanNodeId(obj),
      deserializeStrings(obj["names"]),
      deserializeExpr(obj["projections"]),
      deserializeSingleSource(obj));
```